cmd/erasure-multipart.go

```diff
@@ -987,6 +950,80 @@ func (er erasureObjects) ListObjectParts(ctx context.Context, bucket, object, up
 	return result, nil
 }
 
+func readParts(ctx context.Context, disks []StorageAPI, bucket string, partMetaPaths []string, partNumbers []int, readQuorum int) ([]*ObjectPartInfo, error) {
+	g := errgroup.WithNErrs(len(disks))
+
+	objectPartInfos := make([][]*ObjectPartInfo, len(disks))
+	// Rename file on all underlying storage disks.
+	for index := range disks {
+		index := index
+		g.Go(func() (err error) {
+			if disks[index] == nil {
+				return errDiskNotFound
+			}
+			objectPartInfos[index], err = disks[index].ReadParts(ctx, bucket, partMetaPaths...)
+			return err
+		}, index)
+	}
+
+	if err := reduceReadQuorumErrs(ctx, g.Wait(), objectOpIgnoredErrs, readQuorum); err != nil {
+		return nil, err
+	}
+
+	partInfosInQuorum := make([]*ObjectPartInfo, len(partMetaPaths))
+	partMetaQuorumMap := make(map[string]int, len(partNumbers))
+	for pidx := range partMetaPaths {
+		var pinfo *ObjectPartInfo
+		for idx := range disks {
+			if objectPartInfos[idx] == nil || len(objectPartInfos[idx]) == 0 {
+				continue
+			}
+
+			pinfo = objectPartInfos[idx][pidx]
+			if pinfo == nil {
+				continue
+			}
+
+			if pinfo.ETag != "" {
+				partMetaQuorumMap[pinfo.ETag]++
+			} else {
+				partMetaQuorumMap[partMetaPaths[pidx]]++
+			}
+		}
+		if pinfo == nil {
```

Comment:
It is possible that more than quorum of them had a valid value except the last disk and we'd incorrectly set this part as invalid.

Reply:
I tried to resolve it, please take a look.